onnxruntime/core/providers/cuda/tensor/gather_elements_impl.cu

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "gather_elements_impl.h"
+
+namespace onnxruntime {
+namespace cuda {
+
+template <typename T, typename Tin>
+__global__ void _GatherElementsKernel(
+    const int64_t rank,
+    const T* input_data,
+    const int64_t input_dim_along_axis,
+    const int64_t* input_strides,
+    const Tin* indices_data,
+    const int64_t indices_size,
+    const fast_divmod* indices_strides,
+    const int axis,
+    T* output_data) {
+  CALCULATE_ELEMENTWISE_INDEX_OR_EXIT(indices_index, indices_size);
+  int dim, remain = indices_index;
+  size_t data_idx = 0;
```

Comment:
Should 'dim' be initialized to a known value?

Reply:
Technically it may not be needed as in the first iteration below `dim` gets assigned something meaningful that is then used for processing. Let me know if this line can be made better though...

Reply:
It's cheap to initialize to 0 even though divmod will set it (however it's a bit opaque and requires looking up the divmod implementation to see it's ok to not initialize dim). Optional to change. 

---
In reply to: [342217676](https://github.com/microsoft/onnxruntime/pull/2310#discussion_r342217676) [](ancestors = 342217676)

Reply:
Fixed.